### ISSUE TITLE
tf.nn.sigmoid_cross_entropy_with_logits should support broadcasting

### ISSUE THREAD

**Issue by eli-osherovich** [tensorflow/tensorflow#51967]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): no
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Debian 
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: NA
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.6
- Python version: 3.9
- Bazel version (if compiling from source): NA
- GCC/Compiler version (if compiling from source): NA
- CUDA/cuDNN version: NA
- GPU model and memory: NA

**Describe the current behavior**
`tf.nn.sigmoid_cross_entropy_with_logits` does not support broadcasting. 
This leads to some wrong behavior in certain cases (e.g., keras-team/keras#15247) 
**Describe the expected behavior**
I would expect it to support boradcasting

**[Contributing](https://www.tensorflow.org/community/contribute)**

- Do you want to contribute a PR? (yes/no): yes
- Briefly describe your candidate solution(if contributing):
Add broadcasting for `labels`

**Standalone code to reproduce the issue**
```python
y = tf.random.uniform((10, 1))

# This one works
tf.keras.losses.binary_crossentropy(0.5, y) 

# This one fails
tf.keras.losses.binary_crossentropy(0.5, y, from_logits=True)
```

**Comment by mohantym** [comment]
Hi @eli-osherovich ,I tried to replicate and resolve the issue , But y_true and y_predict should have same number of element inside right. so I tested with example labels and logits here in this [gist.](https://colab.research.google.com/gist/mohantym/83fe136b930375f4a4f7ff8279642017/github_51967.ipynb) Thanks!

**Comment by eli-osherovich** [comment]
@mohantym 
This is exactly the point -- the requirement to have the same number of elements is unjustified in my view and should be removed.  Please have a look at the example where keras fails due to this requirement.

**Comment by mohantym** [comment]
Ok! @sanatmpa1 ! could you please  look at this issue ?

**Comment by sachinprasadhs** [comment]
Based on the comments in the linked PR https://github.com/tensorflow/tensorflow/pull/51968  which was closed, it is not the suggested way to add broadcast in low level API.
You can go ahead and close this issue.
If you are still looking for broadcasting support you can create the new issue in `keras-team/keras` repo.
To know more see;
https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999
Thank you!
